pkg/controllers/work/apply_controller.go

```diff
@@ -463,6 +468,24 @@ func trackResourceAvailability(gvr schema.GroupVersionResource, curObj *unstruct
 	}
 }
 
+func trackPDBAvailability(curObj *unstructured.Unstructured) (ApplyAction, error) {
+	var pdb policyv1.PodDisruptionBudget
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(curObj.Object, &pdb); err != nil {
+		return errorApplyAction, controller.NewUnexpectedBehaviorError(err)
+	}
+	// Check if conditions are up-to-date
+	if poddisruptionbudget.ConditionsAreUpToDate(&pdb) {
```

Comment:
I wonder why this alone is not enough? We just want to make sure that the PDB is taken into effect, we don't care if it blocks eviction or not.